lib/CMakeLists.txt

```diff
@@ -1,7 +1,15 @@
 include(GNUInstallDirs)
 
 add_library(mtkahypar libmtkahypar.cpp)
-target_link_libraries(mtkahypar ${Boost_LIBRARIES} TBB::tbb TBB::tbbmalloc_proxy)
+target_link_libraries(mtkahypar ${Boost_LIBRARIES} TBB::tbb)
+
+# Building shared libs uses malloc_proxy, static uses malloc.
+if(${BUILD_SHARED_LIBS})
+  target_link_libraries(mtkahypar TBB::tbbmalloc_proxy)
+else()
+  target_link_libraries(mtkahypar TBB::tbbmalloc)
+endif()
```

Comment:
I don't know much about TBB, is the assumption correct to use `tbbmalloc` instead of `tbbmalloc_proxy` for a static version? TBB only produces this one on a static build.